mozilla_django_oidc/auth.py

```diff
@@ -105,21 +112,18 @@ def authenticate(self, code=None, state=None):
             user_response.raise_for_status()
             user_info = user_response.json()
             email = user_info.get('email')
-            if not email:
-                return None
 
-            create_user = False
-            try:
-                return self.UserModel.objects.get(email=email)
-            except self.UserModel.MultipleObjectsReturned:
+            # email based filtering
+            users = self.filter_users_by_claims(user_info)
+
+            if len(users) == 1:
+                return users[0]
+            elif len(users) > 1:
                 # In the rare case that two user accounts have the same email address,
                 # log and bail. Randomly selecting one seems really wrong.
                 LOGGER.warn('Multiple users with email address %s.', email)
                 return None
-            except self.UserModel.DoesNotExist:
-                create_user = import_from_settings('OIDC_CREATE_USER', True)
-
-            if create_user:
+            elif import_from_settings('OIDC_CREATE_USER', True):
                 user = self.create_user(email)
```

Comment:
Not completely related but since we are going for a more generic approach I would prefer to allow users to override ``create_user`` and have access to ``claims`` instead of passing just the email.